packages/dds/tree/src/test/time/tableTree.spec.ts

```diff
@@ -0,0 +1,739 @@
+
+				// Setup
+				const localTree: TableTreeDefinition = createTableTree(tableSize, cellValue);
+				setupOperation(localTree);
+
+				// Operation
+				const before = state.timer.now();
+				stackOperation(localTree);
+				const after = state.timer.now();
+
+				// Measure
+				duration = state.timer.toSeconds(before, after);
+			} while (state.recordBatch(duration));
+		},
+		minBatchDurationSeconds,
+		maxBenchmarkDurationSeconds,
+	});
+}
+
+describe("SharedTree execution time", () => {
+	// The value to be set in the cells of the tree.
+	const cellValue = "cellValue";
+	// The test tree's size will be 10*10, 100*100.
+	// Table size 1000 benchmarks removed due to high overhead and unreliable results.
+	const tableSizes = isInPerformanceTestingMode
+		? [10, 100]
+		: // When not measuring perf, use a single smaller data size so the tests run faster.
+			[10];
+
+	// The number of operations to perform on the tree.
+	// Operation counts 1000 removed due to high overhead and unreliable results.
+	const operationCounts = isInPerformanceTestingMode
+		? [10, 100]
+		: // When not measuring perf, use a single smaller data size so the tests run faster.
+			[5];
+	// The maximum duration for each benchmark, in seconds.
+	let maxBenchmarkDurationSeconds: number;
+
+	for (const treeSize of tableSizes) {
+		maxBenchmarkDurationSeconds = treeSize === 100 ? 10 : 5;
+
+		// Filter counts to ensure remove operation do not exceed treeSize
+		const validRemoveCounts = operationCounts.filter((count) => count <= treeSize);
+
+		// Insert-related tests that are not limited by treeSize
+		for (const count of operationCounts) {
+			describe(`Column Insertion`, () => {
+				// Test the execute time of the SharedTree for inserting a column in the middle for a given number of times.
+				runBenchmark({
+					title: `Insert a column in the middle ${count} times`,
+					tableSize: treeSize,
+					cellValue,
+					operation: (tree: TableTreeDefinition) => {
+						const { table } = tree;
+						for (let i = 0; i < count; i++) {
+							const column = new Column({});
+							table.insertColumn({ index: Math.floor(table.columns.length / 2), column });
+						}
+					},
+					maxBenchmarkDurationSeconds,
+				});
+
+				// Test the execute time of undoing insert a column in the middle for a given number of times.
+				runUndoRedoBenchmark({
+					title: `Undo insert the middle column ${count} times`,
+					tableSize: treeSize,
+					cellValue,
+					setupOperation: (tree: TableTreeDefinition) => {
+						const { table } = tree;
+						for (let i = 0; i < count; i++) {
+							const column = new Column({});
+							table.insertColumn({ index: Math.floor(table.columns.length / 2), column });
+						}
+					},
+					stackOperation: (tree: TableTreeDefinition) => {
+						const { undoRedoManager } = tree;
+						for (let i = 0; i < count; i++) {
+							undoRedoManager.undo();
+						}
```

Comment:
Something that occurs to me: to help ensure that these tests are undoing/redoing the changes we think they are, we could add simple sanity-check assertions after we're done going through the operations. In this case, we could add `assert(!undoRedoManager.canUndo)` - in theory, we should have undone all revertibles in the undo stack. If we haven't, then our test might not be undoing the operations we expect it to.

Reply:
Added